daemon/algod/api/server/v2/handlers.go

```diff
@@ -1254,16 +1254,21 @@ func (v2 *Handlers) GetStateProof(ctx echo.Context, round uint64) error {
 	}
 
 	response := generated.StateProofResponse{
-		Message:    protocol.Encode(&tx.Message),
 		StateProof: protocol.Encode(&tx.StateProof),
 	}
 
+	response.Message.BlockHeadersCommitment = tx.Message.BlockHeadersCommitment
+	response.Message.VotersCommitment = tx.Message.VotersCommitment
+	response.Message.LnProvenWeight = tx.Message.LnProvenWeight
+	response.Message.FirstAttestedRound = tx.Message.FirstAttestedRound
+	response.Message.LastAttestedRound = tx.Message.LastAttestedRound
```

Comment:
nit: initialize all at once
```suggestion
		StateProof: protocol.Encode(&tx.StateProof),
		Message: generated.StateProof.Message {
			BlockHeadersCommitment: tx.Message.BlockHeadersCommitment
			VotersCommitment:       tx.Message.VotersCommitment
			LnProvenWeight:         tx.Message.LnProvenWeight
			FirstAttestedRound:     tx.Message.FirstAttestedRound
			LastAttestedRound:      tx.Message.LastAttestedRound
		
		}
	}
```

Reply:
the message is defined as an embedded struct so this code will not work as is.
the idea was to avoid exposing a public structure named `message` since it is not very informative by itself. Rather have it embedded into the state proof response

Reply:
You can still do these sorts of things with embedded structs: https://go.dev/play/p/prtHNzb7xPl

Reply:
But this - does not https://go.dev/play/p/FquXrvHyzeM
Message is generated as embedded:
```
type StateProof struct {

	// Represents the message that the state proofs are attesting to.
	Message struct {

		// The vector commitment root on all light block headers within a state proof interval.
		BlockHeadersCommitment []byte `json:"BlockHeadersCommitment"`
```